File: Laba2/4.cs
using System;
using System.Collections.Generic; // Для List

class Program {
    static void Main() {
        int count = 0;
        List<int> nums = new List<int>();
        Console.Write("Введите числа через пробел: ");
        string input = Console.ReadLine();
        string[] numbers = input.Split(new[] {' '});
        foreach (string numStr in numbers) {
            if (int.TryParse(numStr, out int num)) { // TryParse строку в число
                nums.Add(num);
            }
        }
        foreach (int num in nums) {
            string st = Math.Abs(num).ToString();
            if (st.Length % 2 != 0) {
                count++;
            }
        }
        Console.WriteLine(count);
    }
}

File: Laba2/4.java
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        int count = 0;
        Scanner scanner = new Scanner(System.in);
        System.out.print("Введите числа через пробел: ");
        String input = scanner.nextLine();
        String[] nums = input.split(" "); // Список разделенных элементов
        for (String num : nums) {
            int numb = Integer.parseInt(num); // Строку в число
            String st = Integer.toString(Math.abs(numb)); // Число в строку
            if (st.length() % 2 != 0) {
                count++;
            }
        }
        System.out.println(count);
    }
}
